packages/framework/ai-collab/src/explicit-strategy/debugEvents.ts

```diff
@@ -0,0 +1,294 @@
+	 * to things such as invalid json.
+	 */
+	isLlmResponseValid: TIsLlmResponseValid;
+	/**
+	 * If the `isLlmResponseValid` field value is `true` then this be one of two following values returned by the LLM:
+	 * 1. An edit that can be applied to a SharedTree to further accomplish the user's goal.
+	 * 2. `null` if the LLM decides no more edits are necessary.
+	 *
+	 * If the `isLlmResponseValid` field is `false` then this field will be `undefined`.
+	 */
+	llmGeneratedEdit: TEdit;
+}
+
+/**
+ * An {@link EventFlowDebugEvent} marking the successful application of an edit generated by the LLM to a SharedTree.
+ * @alpha
+ */
+export interface ApplyEditSuccess extends EventFlowDebugEvent {
+	eventName: "APPLIED_EDIT_SUCCESS";
+	eventFlowName: "GENERATE_AND_APPLY_TREE_EDIT";
+	eventFlowStatus: "IN_PROGRESS";
+	/**
+	 * A unique id that will be shared across all debug events that are part of the same event flow.
+	 */
+	eventFlowTraceId: string;
+	/**
+	 * The {@link LlmTreeEdit} generated by the LLM.
+	 */
+	edit: LlmTreeEdit;
+}
+
+/**
+ * An {@link EventFlowDebugEvent} marking the failure of applying an edit generated by the LLM to a SharedTree.
+ * @alpha
+ */
+export interface ApplyEditFailure extends EventFlowDebugEvent {
+	eventName: "APPLIED_EDIT_FAILURE";
+	eventFlowName: "GENERATE_AND_APPLY_TREE_EDIT";
+	eventFlowStatus: "IN_PROGRESS";
+	/**
+	 * A unique id that will be shared across all debug events that are part of the same event flow.
+	 */
+	eventFlowTraceId: string;
+	/**
+	 * The {@link LlmTreeEdit} generated by the LLM.
+	 */
+	edit: LlmTreeEdit;
+	/**
+	 * The error message that was thrown when attempting to apply the edit.
+	 */
+	errorMessage: string;
+	/**
+	 * The total number of errors that have occurred up until this point, not including this error.
+	 */
+	sequentialErrorCount: number;
+}
+
+// #endregion Generate And Apply Tree Edit Debug events ----------------------
+
+// #region Generate Final Review Debug events -------------------------
+
+/**
+ * An {@link EventFlowDebugEvent} marking the initiation of the flow for prompting an LLM to complete a final review of its edits
+ * and determine whether the user's goal was accomplished.
+ * @alpha
+ */
+export interface FinalReviewStarted extends EventFlowDebugEvent {
+	eventName: "FINAL_REVIEW_STARTED";
+	eventFlowName: "FINAL_REVIEW";
+	eventFlowStatus: "STARTED";
+	/**
+	 * The prompt sent to the LLM to complete its final review of the edits its made.
+	 */
+	llmPrompt: string;
+}
+
+/**
+ * An {@link EventFlowDebugEvent} marking the end of the flow for prompting an LLM to complete a final review of its edits
+ * and determine whether the user's goal was accomplished.
+ * @alpha
+ */
+export interface FinalReviewCompleted<
+	TIsLlmResponseValid = boolean,
+	TReviewResponse = TIsLlmResponseValid extends true ? "yes" | "no" : undefined,
+> extends EventFlowDebugEvent {
+	eventName: "FINAL_REVIEW_COMPLETED";
+	eventFlowName: "FINAL_REVIEW";
```

Comment:
Nit: I think we use `EventFlowDebugNames.FINAL_REVIEW` (or corresponding value) for `eventFlowName` across all events?